test/Table_Tests/src/In_Memory/Table_Spec.enso

```diff
@@ -472,7 +472,7 @@ spec =
             c = Column.from_vector 'foo' [My.Data 1 2, My.Data 2 5, My.Data 3 4, My.Data 6 3, Nothing, My.Data 1 0]
             cmp a b = Ordering.compare (a.x-a.y).abs (b.x-b.y).abs
             r = c.sort by=cmp
-            r.to_vector.should_equal [My.Data 1 2, My.Data 3 4, My.Data 1 0, My.Data 2 5, My.Data 6 3, Nothing]
+            r.to_vector.should_equal [Nothing, My.Data 1 2, My.Data 3 4, My.Data 1 0, My.Data 2 5, My.Data 6 3]
```

Comment:
This is not according to the specs that we have in the issue description - Nothing should be at the very end of the sorted vector. This might be a bit tricky for me to implement now. Currently, all the value with the default comparator (both primitive and non-primitive) are sorted at the beginning of the resulting vector, with Nothing and Nans placed at the end of the group.

So the resulting vector has these sections:
```
[<primitive values>, <non-primitive values with Default_Comparator>, <incomparable values with Default_Comparator (NaN, Nothing)>, <other values with different comparators - sorted by FQN of comparators>]
```

TL;DR: Should I revert this change and try harder to implement it, or are we OK with this so far?

cc @JDunkerley

Reply:
cc @radeusgd

Reply:
I guess it's up-to @jdunkerley , as he was the one wanting to have `Nothing` at the end. If this is the requirement, I think we should do it indeed.

I'm fine with doing a fix-up in a separate PR if you think it's better - but would be good to create a ticket for this in that case, as otherwise this will get lost.

IMO though the best way is to fix it now, with some simple workaround. I imagine we could just have a completely special-case branch for `interop.isNull(x)` and put these at the end. Just do a check for null _before_ check for comparator and treat nulls as a _separate_ group from the `DefaultComparator` group. Would that work?